### PR TITLE
Highlight as you type, update control commands

### PR DESCRIPTION
**Pull request by sugarfi**
This PR adds support, when colorize is enabled, for highlighting code as it is typed as well as highlighting result values. As well, the quit, exit, paste, reset and debug commands are changed to #quit, #exit, #paste, #reset, and #debug to allow users to call functions with those same names. The spec was modifed slightly to account for these changes, but the core remains the same and all tests pass.

### CODE REVIEW

File: .icr_NabFp8PkXVhFJjq08tvhQQ.cr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+A = 0
+B = 1
```
Comment (jwoertink) [inline .icr_NabFp8PkXVhFJjq08tvhQQ.cr:2]: These `.icr_` files are supposed to be [ignored](https://github.com/crystal-community/icr/blob/master/.gitignore#L12), but it looks like maybe the name format changed? Or did you add this as part of a test?

File: src/icr/console.cr [hunk 2]
```diff
@@ -26,14 +74,14 @@ module Icr
         # Ctrl+D was pressed, print new line before exit
         puts
         __exit__
-      elsif %w(exit quit).includes?(input.to_s.strip)
+      elsif input.to_s.strip == "#exit" || input.to_s.strip == "#quit"
         __exit__
-      elsif input.to_s.strip == "paste"
+      elsif input.to_s.strip == "#paste"
```
Comment (jwoertink) [inline src/icr/console.cr:79]: I actually dig this change. Using "magic comments" gives us a bit more freedom here. 👍

File: src/icr/executer.cr [hunk 3]
```diff
@@ -39,6 +39,8 @@ module Icr
           io_out.to_s.split(/#{@tmp_file_name}:\d+: /).last.strip +
             "\n" +
             io_error.to_s.strip
+        # Remove "Showing last frame. Use --error-trace for full trace." message and filename
+        #error_message = error_message.split('\n')[4..].join('\n')
```
Comment (jwoertink) [inline src/icr/executer.cr:43]: Was this left in for debugging?

**Review by jwoertink — COMMENTED** [review]
Awesome! I left a few comments, but I think if those `.icr` files aren't meant for testing (which it doesn't look like it), then add those to the ignore as well as that `icr` binary and this should be good to go.

File: .icr_NabFp8PkXVhFJjq08tvhQQ.cr (hunk 1 above)
Comment (sugarfi) [reply .icr_NabFp8PkXVhFJjq08tvhQQ.cr:2]: They were supposed to be ignored, not sure why git added them...

File: src/icr/executer.cr (hunk 3 above)
Comment (sugarfi) [reply src/icr/executer.cr:43]: Yeah, my bad. I meant to take that out...

**Review by jwoertink — COMMENTED** [review]
Super close. I think just remove that `icr` binary, and we're good!

File: src/icr/executer.cr (hunk 3 above)
Comment (jwoertink) [reply src/icr/executer.cr:43]: oh, and I guess this too.

Comment (sugarfi) [reply src/icr/executer.cr]: cool, took it out!

**Review by jwoertink — APPROVED** [review]
Looks great to me! Thanks 👍

**Pull request merged**